File: src/main/scala/com/equalinformation/poc/akka/scala/GreeterActor.scala
package com.equalinformation.poc.akka.scala

import akka.actor.Actor

/**
  * Created by bpupadhyaya on 12/20/15.
  */
object Greeter {
  case object Greet
  case object Done
}

class GreeterActor extends Actor {
  def receive = {
    case Greeter.Greet =>
      println("First actor application")
      sender() ! Greeter.Done
  }
}

File: src/main/scala/com/equalinformation/poc/akka/java/GreeterActor.java
package com.equalinformation.poc.akka.java;

import akka.actor.UntypedActor;

/**
 * Created by bpupadhyaya on 12/20/15.
 */
public class GreeterActor extends UntypedActor {

    public static enum Msg {
        GREET, DONE;
    }

    @Override
    public void onReceive(Object msg) {
        if (msg == Msg.GREET) {
            System.out.println("First actor application");
            getSender().tell(Msg.DONE, getSelf());
        } else
            unhandled(msg);
    }
}
